ONSdigital/dp-search-data-finder
# Remove dependency on search reindex

# What has changed
    
    
    
    
    The following functionality has been removed so that the Search Data Finder no longer has a dependency on the Search Reindex API and so that the new Search Import Tracker can be used, instead of this, to update the Search Reindex API with information about search reindex jobs and their tasks. The Search Data Finder now has a single responsibility, which is to find data that needs to be indexed (or re-indexed) in search.

    The unit and component tests have been updated to reflect the changes.

    The config and README have been updated to remove the SEARCH_REINDEX_URL and SERVICE_AUTH_TOKEN environment variables as these are no longer used.

    # How to review
    
    Check that the code looks correct and that the following tests pass:

    make test

    go test -component

    make lint

    To check the health endpoint locally run this:
    curl -s http://localhost:28000/health | jq

    To check that the reindex-requested event is still being correctly received do as follows:
    - in dp-compose run MongoDB, ElasticSearch and Kafka as follows: docker-compose up -d
    - run zebedee
    - run the Search API (but first run this command to use new ElasticSearch): export ELASTIC_SEARCH_URL="http://localhost:11200"
    - run the Search Reindex API
    - run the Search Data Finder
    - call the POST /search-reindex-jobs endpoint on the Search Reindex API to create a reindex-requested event:
    curl -s http://localhost:25700/search-reindex-jobs | jq

    Look at the Search Data Finder logs and you should see something like this:    

    {
        "created_at": "2022-07-15T16:20:34.800454Z",
        "data": {
            "event": {
            "JobID": "725ba451-bbed-4d9a-8b74-4b4880510491",
            "SearchIndex": "ons1657902034343125",
            "TraceID": "pUVEssoAWdoHGLlI"
            }
        },
        "event": "reindex requested event handler called",
        "namespace": "dp-search-data-finder",
        "severity": 3,
        "trace_id": "pUVEssoAWdoHGLlI"
        }
        {
        "created_at": "2022-07-15T16:20:34.866711Z",
        "data": {
            "first URLs": [
            "/releases/adoption",
            "/releases",
            "/releases/activitysizeandlocation",
            "/economy/grossdomesticproductgdp/timeseries/ihyq",
            "/economy/grossdomesticproductgdp",
            "/economy/regionalaccounts/grossdisposablehouseholdincome/bulletins/regionalgrossdisposablehouseholdincomegdhi/2014-06-04",
            "/economy/regionalaccounts/grossdisposablehouseholdincome/bulletins/regionalgrossdisposablehouseholdincomegdhi/2013-04-24",
            "/economy/regionalaccounts/grossdisposablehouseholdincome/bulletins/regionalgrossdisposablehouseholdincomegdhi/2015-05-27",
            "/economy/regionalaccounts/grossdisposablehouseholdincome",
            "/economy/regionalaccounts"
            ]
        },
        "event": "first 10 URLs retrieved",
        "namespace": "dp-search-data-finder",
        "severity": 3,
        "trace_id": "pUVEssoAWdoHGLlI"
    }

    # Who can review
  
  Anyone but me
